nodejs/week1/homework/src/backend/app.js

```diff
@@ -0,0 +1,71 @@
+const express = require('express');
+const app = express();
+
+// import data here
+const meals = require('./data/meals.json');
+const reservations = require('./data/reservations.json');
+const reviews = require('./data/reviews');
+
+app.get('/', sendGreeting);
+app.get('/meals', listMeals);
+app.get('/reservations', listReservations);
+app.get('/cheap-meals', cheapMealsList);
+app.get('/large-meals', largeMealsList);
+app.get('/meal', listMeal);
+app.get('/reservation', listReservation);
+
+async function sendGreeting(request, response) {
+    response.send('asd');
+}
+
+async function listMeals(request, response) {
+    response.json(getMapAndFilter(meals));
+}
+
+function getMapAndFilter(param) {
+  const mealWithReview = param.map(meal => {
+    const reviewOfEachMeal = reviews.filter(review => review.mealId === meal.id)
+    return {
+      ...meal,
+      reviews: reviewOfEachMeal
+    }
+  })
+  return mealWithReview
+};
+
+async function cheapMealsList(request, response) {
+     const allCheapMeals = meals
+        .filter((meal) => meal.price < 80)
+    response.json(getMapAndFilter(allCheapMeals));
+}
+
+async function largeMealsList(request, response) {
+    const largeMeals = meals
+        .filter((meal) => meal.maxNumberOfGuests > 20)
+        
+    response.json(getMapAndFilter(largeMeals));
+}
+
+async function listMeal(request, response) {
+    let randomMeal = meals[randomValue(meals)]; 
+    const reviewOfRandMeal = meals
+        .filter((meal) => meal.id === randomMeal.id);
+       
+    response.json(getMapAndFilter(reviewOfRandMeal));
```

Comment:
```suggestion
    let randomMeal = meals[randomValue(meals)]; 
    const mealsWithReviews = getMapAndFilter([randomMeal]);
    response.json(mealsWithReviews[0]);
```

Right now the code maps over reviews when it needs to map over meals.